pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV2V3.java

```diff
@@ -61,12 +65,15 @@
   private final byte[] _variableSizeDataBytes;
   private final ByteBuffer _variableSizeData;
   private final Map<String, String> _metadata;
+  // Only V3 has _positionalData
+  private final String[] _positionalData;
 
   /**
    * Construct data table with results. (Server side)
    */
-  public DataTableImplV2(int numRows, DataSchema dataSchema, Map<String, Map<Integer, String>> dictionaryMap,
-      byte[] fixedSizeDataBytes, byte[] variableSizeDataBytes) {
+  public DataTableImplV2V3(int version, int numRows, DataSchema dataSchema,
```

Comment:
Are we passing version number to the constructor so that we can do backward compatibility tests between V2 and V3 ? Other than tests, I don't see why server should decide a version. It should always write the data table with CURRENT_VERSION